### PR TITLE
ADD: New variant Edgebox-ESP-100

### PR DESCRIPTION
**Pull request by peterpanstechland**
Hi, I would like to add [Edgebox-ESP-100](https://files.seeedstudio.com/wiki/edge_box_esp/EdgeBox-ESP-100-User_Manual.pdf) which is an PLC device equipped with ESP32-S3-WROOM-1U

## Changes:
1. added the Edgebox-ESP-100 board config setting to the end of the **"boards.txt"** file 
2. added Edgebox-ESP-100 folder in  **"/varient/Edgebox-ESP-100"**
3. added **"pin_arduino.h"** in  **"/varient/Edgebox-ESP-100"** folder for the pin configurations.

### Hardware Test
I have tested my Pull Request on Arduino-esp32 core v2.0.6 with Edgebox-ESP-100 with successful verify and upload example sketches by using a usb-to-serial tool and connected to the Serial Programming & Debugging port of the Edgebox-ESP-100, The reset button need to be pressed or GPIO0 of the Serial Programming & Debugging port need to be pull down before the board powering up, so the ESP32-S3 module could boot to programming mode on next powering up.

Hardware Wiring:
IMG20230129143749
IMG20230129143613

Uploading Program:

Serial Monitor Result: 

### CI TEST
Passed the CI Test

Thanks.
Peter

### CODE REVIEW

**Comment by peterpanstechland** [comment]
Hi @P-R-O-C-H-Y should I do anything about this for getting it ready to be merge to the next release.
Thanks

File: boards.txt [hunk 1]
```diff
@@ -20649,3 +20649,210 @@ esp32c3m1IKit.menu.EraseFlash.all=Enabled
...
+
+Edgebox-ESP-100.menu.EventsCore.1=Core 1
+Edgebox-ESP-100.menu.EventsCore.1.build.event_core=-DARDUINO_EVENT_RUNNING_CORE=1
+Edgebox-ESP-100.menu.EventsCore.0=Core 0
+Edgebox-ESP-100.menu.EventsCore.0.build.event_core=-DARDUINO_EVENT_RUNNING_CORE=0
+
+Edgebox-ESP-100.menu.USBMode.default=Hardware CDC and JTAG
+Edgebox-ESP-100.menu.USBMode.default.build.usb_mode=0
+Edgebox-ESP-100.menu.USBMode.hwcdc=USB-OTG (TinyUSB)
+Edgebox-ESP-100.menu.USBMode.hwcdc.build.usb_mode=1
```
Comment (P-R-O-C-H-Y) [inline boards.txt:20751]: Can you change it back to the default values as is for ESP32-S3 dev module?
So please switch 0/1 and change `Edgebox-ESP-100.build.usb_mode=0` to 1.
It is better to keep these settings the same. This won't have any impact on your board definition.

File: boards.txt [hunk 2]
```diff
@@ -20649,3 +20649,210 @@ esp32c3m1IKit.menu.EraseFlash.all=Enabled
...
+
+Edgebox-ESP-100.build.tarch=xtensa
+Edgebox-ESP-100.build.bootloader_addr=0x0
+Edgebox-ESP-100.build.target=esp32s3
+Edgebox-ESP-100.build.mcu=esp32s3
+Edgebox-ESP-100.build.core=esp32
+Edgebox-ESP-100.build.variant=Edgebox-ESP-100
+Edgebox-ESP-100.build.board=Edgebox-ESP-100
+
+Edgebox-ESP-100.build.usb_mode=0
```
Comment (P-R-O-C-H-Y) [inline boards.txt:20680]: Change `Edgebox-ESP-100.build.usb_mode=0` to 1 to match changes in next comment.

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
Hi @peterpanstechland, please take a look on my comment. Just one small thing, after its fixed we can merge :)

File: boards.txt (hunk 1 above)
Comment (peterpanstechland) [reply boards.txt]: The latest commit should have the changes as request, Thanks.

File: boards.txt (hunk 2 above)
Comment (peterpanstechland) [reply boards.txt]: The latest commit should have the changes as request, Thanks.

**Comment by peterpanstechland** [comment]
Hi @P-R-O-C-H-Y 
Sorry that I think I have screwed up the laster merge as I am not very familiar with the Github, Where I believe I have should have synced the `espressif:master` into my Repo before I commit to resolve the conflict. So for now seems like my latest commit(resolve the conflict) contains all the previous changes from other contributor during the period between my first pr and latest commit, Will this create any issues or should be okey for now. 
Thanks

**Review by P-R-O-C-H-Y — APPROVED** [review]
No worries, it looks good. LGTM!

**Comment by VojtechBartoska** [comment]
PTAL on last comment @me-no-dev, thanks

**Comment by me-no-dev** [comment]
@d4b005t have a look here: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/jtag-debugging/index.html

**Pull request merged**